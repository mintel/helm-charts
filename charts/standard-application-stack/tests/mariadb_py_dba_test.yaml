suite: Test MariaDB py-dba configs
templates:
  - mariadb-py-dba.yaml
release:
  namespace: test-namespace
tests:
  - it: extraUsers adds job and configmap
    set:
      global.name: test-app
      mariadb:
        enabled: true
        extraUsers:
          enabled: true
          users:
          - username: user1
            database: test_db
            tables:
            - table: table1
              permissions: ["ALL"]

    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: ConfigMap
        documentIndex: 1
      - isKind:
          of: Job
        documentIndex: 0

  - it: adds correct config to configmap
    set:
      global.name: test-app
      mariadb:
        enabled: true
        extraUsers:
          enabled: true
          users:
          - username: user1
            database: test_db
            tables:
            - table: table1
              permissions: ["ALL"]

    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - equal:
          path: data.py-dba-config
          value: |
            users:
            - database: test_db
              tables:
              - permissions:
                - ALL
                table: table1
              username: user1
        documentIndex: 1
