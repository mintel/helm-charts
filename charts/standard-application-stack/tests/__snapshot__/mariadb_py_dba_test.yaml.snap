adds correct config to configmap:
  1: |
    apiVersion: batch/v1
    kind: Job
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/hook: PreSync
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: local
        app.mintel.com/region: local
        name: test-app
      name: py-dba-test-app-mariadb
      namespace: test-namespace
    spec:
      template:
        spec:
          containers:
            - env:
                - name: LOG_LEVEL
                  value: INFO
                - name: CONFIG_FILE
                  value: /etc/config/py-dba-config
                - name: DB_ENGINE
                  value: mariadb
                - name: USER_SECRET_PREFIX
                  value: test-namespace/test-app/mariadb
              envFrom:
                - secretRef:
                    name: test-app-mariadb
              image: 551844124467.dkr.ecr.us-east-2.amazonaws.com/gitlab/mintel/satoshi/tools/py-dba:v0.2.0
              imagePullPolicy: IfNotPresent
              name: main
              resources:
                limits:
                  memory: 64Mi
                requests:
                  cpu: 100m
                  memory: 32Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                runAsNonRoot: true
                runAsUser: 1000
              volumeMounts:
                - mountPath: /etc/config
                  name: config-volume
          restartPolicy: Never
          securityContext:
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: test-app
          volumes:
            - configMap:
                name: test-app-py-dba-config
              name: config-volume
      ttlSecondsAfterFinished: 600
  2: |
    apiVersion: v1
    data:
      py-dba-config: |
        users:
        - database: test_db
          tables:
          - permissions:
            - ALL
            table: table1
          username: user1
    kind: ConfigMap
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: local
        app.mintel.com/region: local
        name: test-app
      name: test-app-py-dba-config
      namespace: test-namespace
extraUsers adds job and configmap:
  1: |
    apiVersion: batch/v1
    kind: Job
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/hook: PreSync
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: local
        app.mintel.com/region: local
        name: test-app
      name: py-dba-test-app-mariadb
      namespace: test-namespace
    spec:
      template:
        spec:
          containers:
            - env:
                - name: LOG_LEVEL
                  value: INFO
                - name: CONFIG_FILE
                  value: /etc/config/py-dba-config
                - name: DB_ENGINE
                  value: mariadb
                - name: USER_SECRET_PREFIX
                  value: test-namespace/test-app/mariadb
              envFrom:
                - secretRef:
                    name: test-app-mariadb
              image: 551844124467.dkr.ecr.us-east-2.amazonaws.com/gitlab/mintel/satoshi/tools/py-dba:v0.2.0
              imagePullPolicy: IfNotPresent
              name: main
              resources:
                limits:
                  memory: 64Mi
                requests:
                  cpu: 100m
                  memory: 32Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                runAsNonRoot: true
                runAsUser: 1000
              volumeMounts:
                - mountPath: /etc/config
                  name: config-volume
          restartPolicy: Never
          securityContext:
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: test-app
          volumes:
            - configMap:
                name: test-app-py-dba-config
              name: config-volume
      ttlSecondsAfterFinished: 600
  2: |
    apiVersion: v1
    data:
      py-dba-config: |
        users:
        - database: test_db
          tables:
          - permissions:
            - ALL
            table: table1
          username: user1
    kind: ConfigMap
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: local
        app.mintel.com/region: local
        name: test-app
      name: test-app-py-dba-config
      namespace: test-namespace
